storeapi/go-wrapper/microsoftstore/store.go

```diff
@@ -26,3 +28,29 @@ func findWorkspaceRoot() (string, error) {
 		}
 	}
 }
+
+// checkError inspects the values of hres and err to determine what kind of error we have, if any, according to the rules of syscall/dll_windows.go.
+func checkError(hres int64, err error) (int64, error) {
+	// From syscall/dll_windows.go (*Proc).Call doc:
+	// > Callers must inspect the primary return value to decide whether an
+	//   error occurred [...] before consulting the error.
+	if e := NewStoreAPIError(hres); e != nil {
+		return hres, fmt.Errorf("storeApi returned error code %d: %w", hres, e)
+	}
+
+	if err == nil {
+		return hres, nil
+	}
+
+	var target syscall.Errno
+	if b := errors.As(err, &target); !b {
+		// Supposedly unrechable: proc.Call must always return a syscall.Errno
+		return hres, err
+	}
+
+	if target != syscall.Errno(0) {
+		return hres, fmt.Errorf("failed syscall to storeApi: %v (syscall errno %d)", target, err)
+	}
+
+	return hres, nil
```

Comment:
This looks weird to me, so we have a non nil error, which was not an errno, but is Errno(0), so it’s a successfull error? :) It seems that the API can’t send that (at least, not from the tests), but maybe give it a good comment?

Reply:
According to the syscall documentation, there is no possibility of nil error, the `err` return value is always constructed with the result of `GetLastError()` which could have been set by something completely unrelated to our code sometime in the past, as well as it could be `ERROR_SUCCESS` which is the `Errno(0)`. :)

If the act of calling the API fails (not the function we're calling, but the attempt to call it), then we'd get a meaningful `syscall.Errno` object via the `err` parameter, certainly no `Errno(0)`, related to the actual failure (like a function not found in this DLL or whatever).

Since our implementation of the store API doesn't touch errno the call should return `hres` between -128 and 0 plus garbage in the `err` argument, we only care about the `hres` in this case (the begining of the function).

`err` should never be nil nor something other than `syscall.Errno`, so I'm not sure about the actual reason to have added this `error.As` check.